src/jit/codegencommon.cpp

```diff
@@ -10317,10 +10333,23 @@ void CodeGen::genFuncletProlog(BasicBlock* block)
 
     compiler->unwindBegProlog();
 
+    inst_RV(INS_push, REG_FPBASE, TYP_REF);
+    compiler->unwindPush(REG_FPBASE);
+
     // TODO Save callee-saved registers
 
     // This is the end of the OS-reported prolog for purposes of unwinding
     compiler->unwindEndProlog();
+
+    // Restore frame pointer
+    // TODO We will need changes here when we introduce PSPSym
+    inst_RV_RV(INS_mov, REG_FPBASE, REG_ARG_0);
```

Comment:
It does not get to `CallEHFunclet`. It seems like it crashes when processing some unmanaged frame in `UnwindManagedExceptionPass2()`.

I tried `.cfi_def_cfa_offset 4` here(https://github.com/wateret/coreclr/commit/8e7019bb43726ca59a965761f1222852e2bda4be) but did not make changes unfortunately.